### PR TITLE
Refactor inventory mgmt

### PR DESCRIPTION
**Pull request by codestreamtest**
Finally got around to this refactor. I'm sure I got it perfect on the first try. ;-)

### CODE REVIEW

File: foo.js [hunk 1]
```diff
@@ -13,6 +13,22 @@ import { Command, createCommandDecorator, Strings } from "./system";
 const commandRegistry: Command[] = [];
 const command = createCommandDecorator(commandRegistry);
 
+export interface ShowReviewDiffCommandArgs {
+	reviewId: string;
+	checkpoint: CSReviewCheckpoint;
+	repoId: string;
+	path: string;
+}
```
Comment (planteater) [inline foo.js:21]: Doesn't look quite right.

**Comment by planteater** [comment]
This is a comment<div>with a line break.</div>

**Comment by planteater** [comment]
Another comment<div>with another line break</div>

File: foo.js [hunk 2]
```diff
@@ -13,6 +13,22 @@ import { Command, createCommandDecorator, Strings } from "./system";
...
+
+export interface ShowReviewLocalDiffCommandArgs {
+	repoId: string;
```
Comment (planteater) [inline foo.js:24]: Lets fix the foo.

File: tracking.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { Container } from "./container";
+
+export class ProtocolHandler implements UriHandler {
+	private disposable: Disposable;
+
+	constructor() {
+		this.disposable = window.registerUriHandler(this);
+	}
+
+	async handleUri(uri: Uri) {
```
Comment (planteater) [inline tracking.js:12]: This is a comment on lines 12 - 14

File: foo.js (hunk 2 above)
Comment (planteater) [reply foo.js:24]: reply to the foo

Comment (codestreamtest) [reply foo.js:24]: no more foo for you

**Comment by codestreamtest** [comment]
I like PRs

File: foo.js [hunk 4]
```diff
@@ -13,6 +13,22 @@ import { Command, createCommandDecorator, Strings } from "./system";
...
+	path: string;
+	editingReviewId?: string;
+	includeSaved: boolean;
```
Comment (bcanzanella) [inline foo.js:27]: 2417

File: foo.js [hunk 5]
```diff
@@ -13,6 +13,22 @@ import { Command, createCommandDecorator, Strings } from "./system";
...
 const command = createCommandDecorator(commandRegistry);
 
+export interface ShowReviewDiffCommandArgs {
+	reviewId: string;
+	checkpoint: CSReviewCheckpoint;
+	repoId: string;
+	path: string;
+}
+
+export interface ShowReviewLocalDiffCommandArgs {
```
Comment (planteater) [inline foo.js:23]: hot messness

File: foo.js [hunk 6]
```diff
@@ -13,6 +13,22 @@ import { Command, createCommandDecorator, Strings } from "./system";
...
+	repoId: string;
+	path: string;
+	editingReviewId?: string;
+	includeSaved: boolean;
+	includeStaged: boolean;
+	baseSha: string;
```
Comment (planteater) [inline foo.js:29]: i can fix all ur codz

File: tracking.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+		this.disposable = window.registerUriHandler(this);
+	}
+
+	async beerHandleUri(uri: Uri) {
+		await Container.webview.handleProtocol(uri);
+	}
+
+	dispose(): void {
+		this.disposable && this.disposable.dispose();
+	}
```
Comment (planteater) [inline tracking.js:18]: how about lines 16-18

Comment (planteater) [reply tracking.js:18]: Reply to "16-18"

**Comment by planteater** [comment]
Telem -  mainline comment

File: tracking.js (hunk 7 above)
Comment (planteater) [reply tracking.js:18]: Telem - review reply

File: foo.js [hunk 8]
```diff
@@ -10,9 +10,25 @@ import { Container } from "./container";
 import { Logger } from "./logger";
 import { Command, createCommandDecorator, Strings } from "./system";
 
-const commandRegistry: Command[] = [];
+const commandCenter: Command[] = [];
 const command = createCommandDecorator(commandRegistry);
 
+export interface ShowReviewDiffCommandArgs {
+	reviewId: string;
+	checkpoint: CSReviewCheckpoint;
```
Comment (planteater) [inline foo.js:18]: Telem - single comment

File: foo.js [hunk 9]
```diff
@@ -10,9 +10,25 @@ import { Container } from "./container";
...
+	repoId: string;
+	path: string;
```
Comment (planteater) [inline foo.js:20]: Telem - start review

File: tracking.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+export class ProtocolHandler implements UriHandler {
+	private disposable: Disposable;
+
+	constructor() {
+		this.disposable = window.registerUriHandler(this);
+	}
+
+	async beerHandleUri(uri: Uri) {
+		await Container.webview.handleProtocol(uri);
+	}
```
Comment (planteater) [inline tracking.js:14]: Telem - 2nd in review

File: tracking.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+	dispose(): void {
+		this.disposable && this.disposable.dispose();
```
Comment (planteater) [inline tracking.js:17]: 3rd in rev

File: inventory.js [hunk 12]
```diff
@@ -0,0 +1,300 @@
...
+	TextEditorRevealType,
+	Uri,
+	ViewColumn,
+	window,
+	workspace
+} from "vscode";
+import { Position as LspPosition, Range as LspRange } from "vscode-languageclient";
+import { configuration } from "../configuration";
+import { BuiltInCommands } from "../constants";
+import { Container } from "../container";
```
Comment (planteater) [inline inventory.js:20]: start another review

File: inventory.js [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
+	TextDocumentShowOptions,
+	TextEditor,
+	TextEditorRevealType,
+	Uri,
+	ViewColumn,
+	window,
+	workspace
+} from "vscode";
+import { Position as LspPosition, Range as LspRange } from "vscode-languageclient";
+import { configuration } from "../configuration";
```
Comment (planteater) [inline inventory.js:18]: start start

File: inventory.js [hunk 14]
```diff
@@ -0,0 +1,300 @@
...
+import { Logger } from "../logger";
+
+const highlightDecorationType = window.createTextEditorDecorationType({
+	rangeBehavior: DecorationRangeBehavior.OpenOpen,
+	backgroundColor: "rgba(127, 127, 127, 0.4)"
+});
+
+declare global {
+	// Workaround for https://stackoverflow.com/questions/56248618/how-to-check-if-an-object-is-a-readonly-array-in-typescript
+	// https://github.com/microsoft/TypeScript/issues/17002
```
Comment (planteater) [inline inventory.js:30]: startin da review

**Comment by planteater** [comment]
no in changeset 1

```
	reviewId: string;

```
								
[foo.js](https://github.com/codestreamtest/beerfridge/blob/426c916107e3556e85ba3b5137d224aed52c40e8/foo.js#L39) (Line 39)

File: foo.js (hunk 9 above)
Comment (planteater) [inline foo.js:20]: Not so great

**Review by planteater — APPROVED** [review]
Looks awesome!

File: foo.js (hunk 9 above)
Comment (planteater) [inline foo.js:20]: Doesn't look quite right.

**Review by planteater — APPROVED** [review]
Looks good!

File: foo.js [hunk 15]
```diff
@@ -10,9 +10,25 @@ import { Container } from "./container";
...
+}
+
+export interface ShowReviewLocalDiffCommandArgs {
+	repoId: string;
+	path: string;
+	editingReviewId?: string;
+	includeSaved: boolean;
+	includeStaged: boolean;
```
Comment (marcelofarias) [inline foo.js:28]: pr comment from intellij

File: inventory.js (hunk 12 above)
Comment (planteater) [inline inventory.js:20]: fdasdfd asdf asdf asdf sadf sdaf sdaf sadf sdf

File: inventory.js [hunk 16]
```diff
@@ -0,0 +1,300 @@
...
+import { EditorMetrics, EditorSelection } from "@codestream/protocols/webview";
+import {
+	commands,
+	DecorationRangeBehavior,
+	Position,
+	Range,
+	Selection,
+	TextDocumentShowOptions,
+	TextEditor,
+	TextEditorRevealType,
```
Comment (planteater) [inline inventory.js:11]: asdf adddd

**Review by planteater — COMMENTED** [review]
done

**Pull request closed without merging**